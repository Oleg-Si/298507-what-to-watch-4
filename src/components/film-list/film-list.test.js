import React from 'react';
import renderer from 'react-test-renderer';
import FilmList from './film-list.jsx';

const films = [
  {
    id: 0,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`
  },
  {
    id: 2,
    title: `Macbeth`,
    img: `img/macbeth.jpg`
  },
  {
    id: 3,
    title: `Aviator`,
    img: `img/aviator.jpg`
  }
];

const onFilmCardTitleClick = () => {};

it(`Проверяет снепшот компонента FilmList`, () => {
  const tree = renderer.create(
      <FilmList
        films={films}
        onFilmCardTitleClick={onFilmCardTitleClick}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
