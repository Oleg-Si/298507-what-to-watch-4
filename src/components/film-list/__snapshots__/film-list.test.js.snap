// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяет снепшот компонента FilmList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="preloader"
      style={
        Object {
          "left": "50%",
          "marginLeft": "-32px",
          "marginTop": "-32px",
          "position": "absolute",
          "top": "50%",
          "zIndex": "2",
        }
      }
    >
      <svg
        height="64px"
        viewBox="0 0 128 128"
        width="64px"
        xmlSpace="preserve"
      >
        <rect
          fill="#FFFFFF"
          height="100%"
          width="100%"
          x="0"
          y="0"
        />
        <g>
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="16"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="14.344"
            transform="rotate(45 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="12.531"
            transform="rotate(90 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="10.75"
            transform="rotate(135 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="10.063"
            transform="rotate(180 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="8.063"
            transform="rotate(225 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="6.438"
            transform="rotate(270 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="5.375"
            transform="rotate(315 64 64)"
          />
          <animateTransform
            attributeName="transform"
            calcMode="discrete"
            dur="720ms"
            repeatCount="indefinite"
            type="rotate"
            values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
          />
        </g>
      </svg>
    </div>
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        poster="img/aviator.jpg"
        src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/0"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="preloader"
      style={
        Object {
          "left": "50%",
          "marginLeft": "-32px",
          "marginTop": "-32px",
          "position": "absolute",
          "top": "50%",
          "zIndex": "2",
        }
      }
    >
      <svg
        height="64px"
        viewBox="0 0 128 128"
        width="64px"
        xmlSpace="preserve"
      >
        <rect
          fill="#FFFFFF"
          height="100%"
          width="100%"
          x="0"
          y="0"
        />
        <g>
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="16"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="14.344"
            transform="rotate(45 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="12.531"
            transform="rotate(90 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="10.75"
            transform="rotate(135 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="10.063"
            transform="rotate(180 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="8.063"
            transform="rotate(225 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="6.438"
            transform="rotate(270 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="5.375"
            transform="rotate(315 64 64)"
          />
          <animateTransform
            attributeName="transform"
            calcMode="discrete"
            dur="720ms"
            repeatCount="indefinite"
            type="rotate"
            values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
          />
        </g>
      </svg>
    </div>
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        poster="img/macbeth.jpg"
        src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="preloader"
      style={
        Object {
          "left": "50%",
          "marginLeft": "-32px",
          "marginTop": "-32px",
          "position": "absolute",
          "top": "50%",
          "zIndex": "2",
        }
      }
    >
      <svg
        height="64px"
        viewBox="0 0 128 128"
        width="64px"
        xmlSpace="preserve"
      >
        <rect
          fill="#FFFFFF"
          height="100%"
          width="100%"
          x="0"
          y="0"
        />
        <g>
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="16"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="14.344"
            transform="rotate(45 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="12.531"
            transform="rotate(90 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="10.75"
            transform="rotate(135 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="10.063"
            transform="rotate(180 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="8.063"
            transform="rotate(225 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="6.438"
            transform="rotate(270 64 64)"
          />
          <circle
            cx="16"
            cy="64"
            fill="#000000"
            fillOpacity="1"
            r="5.375"
            transform="rotate(315 64 64)"
          />
          <animateTransform
            attributeName="transform"
            calcMode="discrete"
            dur="720ms"
            repeatCount="indefinite"
            type="rotate"
            values="0 64 64;315 64 64;270 64 64;225 64 64;180 64 64;135 64 64;90 64 64;45 64 64"
          />
        </g>
      </svg>
    </div>
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        poster="img/snatch.jpg"
        src="https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/2"
        onClick={[Function]}
      >
        Snatch
      </a>
    </h3>
  </article>
</div>
`;
